src/api.rs

```diff
@@ -1438,3 +1507,55 @@ impl Handle {
         Ok(())
     }
 }
+
+// Struct wrapping an [IAudioEffectsManager](https://learn.microsoft.com/en-us/windows/win32/api/audioclient/nn-audioclient-iaudioeffectsmanager).
+pub struct AudioEffectsManager {
+    manager: IAudioEffectsManager,
+}
+
+impl AudioEffectsManager {
+    /// Gets the current list of audio effects for the associated audio stream.
+    pub fn get_audio_effects(&self) -> WasapiRes<Option<Vec<AUDIO_EFFECT>>> {
+        let mut audio_effects: *mut AUDIO_EFFECT = std::ptr::null_mut();
+        let mut num_effects: u32 = 0;
+
+        unsafe {
+            self.manager
+                .GetAudioEffects(&mut audio_effects, &mut num_effects)?;
+        }
+
+        if num_effects > 0 {
+            let effects_slice =
+                unsafe { slice::from_raw_parts(audio_effects, num_effects as usize) };
```

Comment:
Why go via a slice and not make a vec right away?
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_raw_parts

Reply:
According to the [IAudioEffectsManager::GetAudioEffects API documentation](https://learn.microsoft.com/en-us/windows/win32/api/audioclient/nf-audioclient-iaudioeffectsmanager-getaudioeffects#remarks), after copying the data to a Vec, I added a `CoTaskMemFree` call to deallocate the AUDIO_EFFECT array.

Reply:
Ok, thanks for looking at this, this seems to be the best way to handle this.